src/atldld/sync.py

```diff
@@ -519,6 +520,174 @@ def corners_sagittal(r, dataset_id):
     return section_number_corner_mode, image_id_corner_mode
 
 
+def get_transform_parallel(
+    slice_coordinate: float,
+    matrix_2d: np.ndarray,
+    matrix_3d: np.ndarray,
+    axis: str = "coronal",
+    ds_r: int = 1,
+    ds_i: int = 0,
+) -> DisplacementField:
+    """Compute displacement field between the reference space and the image.
+
+    Parameters
+    ----------
+    slice_coordainte : float
+        Value of the `axis` coordinate at which the image was sliced.
+    matrix_2d : np.array
+        Matrix of shape `(2, 3)` representing a 2D linear transformation.
+    matrix_3d : np.array
+        Matrix of shape `(3, 4)` representing a 3D linear transformation.
+    axis : str, {"coronal", "sagittal", "transverse"}
+        Axis along which the slice was made.
+    ds_r : int
+        Downscaling of the reference space grid. If set to 1 no
+        downsampling takes place. The higher the value the smaller the grid
+        in the reference space and the faster the matrix multiplication.
+    ds_i : int
+        The downloaded image will have both the height and the width
+        downsampled by `2 ** ds_i`.
+
+    Returns
+    -------
+    DisplacementField
+        Displacement field representing the transformation between the
+        reference space and the image. Note that one can directly use it
+        to register raw histological images to the reference space.
+    """
+    refspace = (  # order matters
+        ("coronal", 13200),
+        ("transverse", 8000),
+        ("sagittal", 11400),
+    )
+    axis_fixed = [i for i, a in enumerate(refspace) if a[0] == axis][0]
+    axes_variable = [i for i, a in enumerate(refspace) if a[0] != axis]
+
+    grid_shape = [refspace[i][1] // ds_r for i in axes_variable]
+    n_pixels = np.prod(grid_shape)
+
+    coords_ref = np.ones((4, n_pixels))
```

Comment:
I guess in theory one could do it without flattening. However, IMO flattening across the pixels makes the code more readable and the matrix multiplications are easier to read.

However, I would be willing to change it in case there is a major performance benefit

Reply:
I will resolve, feel free to add comments/unresolve

Reply:
Ha, that's funny, it must be personal, but I find it easier to read when it's not flattened 🙈 Maybe you prefer thinking in terms of 2D matrices? Anyway, doesn't matter so much :)